azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/auth/authentication_provider.py

```diff
@@ -22,6 +22,7 @@ def __init__(self, hostname, device_id, module_id=None):
     def get_current_sas_token(self):
         pass
 
+    # @abc.abstractstaticmethod
```

Comment:
Ideally it would be @abc.abstractstaticmethod, since parse is not a regular method, but a static method (does not have self, or cls as a parameter). That is to say, as is, with the @abc.abstractmethod, this is incorrect. Unfortunately, @abc.abstractstaticmethod does not exist in 2.7. At some we'll have to fix this, but it seemed low priority, so I left it alone for now.

Reply:
i tried this. It did not work  for all versions of python. I think I tried another way also. Using 2 decorators `@staticmethod`
 and `@abstractmethod` but that too did not work. I think both of these work in python 3.2  up and above.

---
In reply to: [252901525](https://github.com/Azure/azure-iot-sdk-python-preview/pull/45#discussion_r252901525) [](ancestors = 252901525)